### PR TITLE
updated rp2 docs-intro.rst file

### PR DESCRIPTION
**Pull request by NitiKaur**
Updated docs with previous changes and changes to the ADC, PWM, Timer, Watchdog and deep-sleep class in the quickref

### CODE REVIEW

File: docs/rp2/general.rst [hunk 1]
```diff
@@ -10,9 +10,37 @@ the RP2040.
...
+# On-chip programmable LDO to generate core voltage
+# 2 on-chip PLLs to generate USB and core clocks
+# 30 GPIO pins, of which 4 can optionally be used as analog inputs
+
+The peripherals include 2 UARTs, 2 SPI controllers, 2 I2C contollers, 16 PWM channels, USB 1.1 controller and PHY, with host and device support, 8 PIO state machines.
+
+RP2040 is the debut microcontroller from Raspberry Pi from the Raspberry Pi Foundation.
+
+It has large on-chip memory, symmetric dual-core processor complex, deterministic bus fabric, and rich peripheral set augmented with  unique Programmable I/O (PIO) subsystem, providing professional users with unrivalled power and flexibility. It has the lowest possible barrier to entry for beginners because of its polished MicroPython port and a UF2 bootloader in ROM.
+<<<<<<< HEAD
```
Comment (tjvr) [inline docs/rp2/general.rst:33]: Looks like a git conflict resolution marker snuck in here by mistake!

**Comment by mattytrentini** [comment]
General feedback: 

- Please wrap lines at 80 columns, some are very long
- The quickref should have only *brief* technical details followed by some terse code examples

File: docs/rp2/general.rst [hunk 2]
```diff
@@ -10,9 +10,28 @@ the RP2040.
...
+
+The peripherals include 2 UARTs, 2 SPI controllers, 2 I2C contollers, 16 PWM channels, USB 1.1 controller and PHY, with host and device support, 8 PIO state machines.
+
+RP2040 is the debut microcontroller from Raspberry Pi from the Raspberry Pi Foundation.
+
+It has large on-chip memory, symmetric dual-core processor complex, deterministic bus fabric, and rich peripheral set augmented with  unique Programmable I/O (PIO) subsystem, providing professional users with unrivalled power and flexibility. It has the lowest possible barrier to entry for beginners because of its polished MicroPython port and a UF2 bootloader in ROM.
+
+RP2040 is a stateless device, with support for cached execute-in-place from external QSPI memory. This gives us the flexibility to choose the appropriate density of non-volatile storage for the application, and the additional benefit of low pricing of commodity Flash parts.
+
+Manufactured on a modern 40nm process node, the RP2040 delivers high performance, low dynamic power consumption, and low leakage, with support for extended-duration operation on battery power through a variety of low-power operations.
```
Comment (mattytrentini) [inline docs/rp2/general.rst:36]: The last four paragraphs are probably best removed; there's enough information above.

File: docs/rp2/quickref.rst [hunk 3]
```diff
@@ -112,7 +121,7 @@ Use the ``machine.PWM`` class::
 ADC (analog to digital conversion)
 ----------------------------------
 
-How does the ADC module work?
+RP2040 has five ADC channels in total, four of which are 12-bit SAR based ADCs: GP26, GP27, GP28 and GP29. The input signal for ADC0, ADC1 and ADC2 can be connected with GP26,GP27,GP28. The fourth is used to measure the VSYS voltage on the board.The standard ADC range is 0-3.3V.
```
Comment (mattytrentini) [inline docs/rp2/quickref.rst:124]: This is good information but we should clarify how the fifth ADC channel can be used.

File: docs/rp2/quickref.rst [hunk 4]
```diff
@@ -221,7 +230,7 @@ See :ref:`machine.WDT <machine.WDT>`. ::
 Deep-sleep mode
```
Comment (mattytrentini) [inline docs/rp2/quickref.rst:230]: Lets remove the deep sleep section for now. 

The current implementation uses lightsleep and then resets the device which is not ideal behaviour. Let's fix the implementation first and then update the docs.

File: docs/rp2/quickref.rst [hunk 5]
```diff
@@ -59,7 +59,8 @@ Use the :mod:`time <utime>` module::
 Timers
```
Comment (mattytrentini) [inline docs/rp2/quickref.rst:59]: We need a code example here how to use the timers.
Can look at the ESP32 & ESP8266 quickref's but ensure that the Timer ID is -1 for rp2.

**Pull request closed without merging**